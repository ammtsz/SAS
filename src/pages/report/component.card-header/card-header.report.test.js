import React from "react";
import CardHeader from "./card-header.report.component";

import { render, screen, cleanup } from "@testing-library/react";

describe("Component report-page <CardHeader />", () => {
  const mockCategory = { id: 1, name: "Sports" };

  describe("Report created right after a quiz is finished", () => {
    const mockProps = {
      review: false,
      category: mockCategory,
    };

    let component;

    beforeEach(() => {
      component = render(<CardHeader {...mockProps} />);
    });

    afterEach(() => {
      cleanup();
    });

    it("should render CardHeader", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });

    it("should contain the 'congratulation' text", () => {
      expect(screen.getByText("Congratulation!")).toBeInTheDocument();
      expect(screen.getByText("You finished the quiz")).toBeInTheDocument();
    });

    it("should NOT contain the category-name text", () => {
      expect(screen.queryByText(mockCategory.name)).toEqual(null);
    });
  });

  describe("Report requested from categories page", () => {
    const mockProps = {
      review: true,
      category: mockCategory,
    };

    let component;

    beforeEach(() => {
      component = render(<CardHeader {...mockProps} />);
    });

    afterEach(() => {
      cleanup();
    });

    it("should render CardHeader", () => {
      expect(component.asFragment()).toMatchSnapshot();
    });

    it("should contain the category-name text", () => {
      expect(screen.getByText(mockCategory.name)).toBeInTheDocument();
    });

    it("should NOT contain the 'congratulation' text", () => {
      expect(screen.queryByText("Congratulation!")).toEqual(null);
      expect(screen.queryByText("You finished the quiz")).toEqual(null);
    });
  });
});
