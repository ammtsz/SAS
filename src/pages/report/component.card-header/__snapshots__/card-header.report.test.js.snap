// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component report-page <CardHeader /> Report created right after a quiz is finished should render CardHeader 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ddYKHZ"
  >
    <img
      class="sc-gsTCUz iydVwP"
      src="mascot.svg"
    />
    <div
      class="sc-dlfnbm kjwOKs"
    >
      <h2
        class="sc-hKgILt hJcEo"
      >
        Congratulation!
      </h2>
      <p
        class="sc-eCssSg eOtUap"
      >
        You finished the quiz
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component report-page <CardHeader /> Report requested from categories page should render CardHeader 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ddYKHZ"
  >
    <img
      class="sc-gsTCUz iydVwP"
      src="mascot.svg"
    />
    <div
      class="sc-dlfnbm kjwOKs"
    >
      <h2
        class="sc-hKgILt hJcEo"
      >
        Sports
      </h2>
      <p
        class="sc-eCssSg eOtUap"
      />
    </div>
  </div>
</DocumentFragment>
`;
